### PR TITLE
S4426 FN: On .Net 7 when using ECDsaOpenSsl

### LINKED ISSUE #6164: S4426 FN: On .Net 7 when using ECDsaOpenSsl
### Description

The rule S4426 does not raise issues for `ECDsaOpenSsl` when running on .Net 7 runtime.

### Repro steps

```
var ec3 = new ECDsaOpenSsl();
ec3.GenerateKey(ECCurve.NamedCurves.brainpoolP192t1); // Noncompliant {{Use a key length of at least 224 bits for EC cipher algorithm.}}
```

### Expected behavior

An issue should be raised.

### Actual behavior

No issue raised.

### Known workarounds

Please provide a description of any known workarounds.

### PR DESCRIPTION
**Pull request by cristian-ambrosini-sonarsource**
Fixes #6164

### CODE REVIEW

File: analyzers/src/SonarAnalyzer.Common/Helpers/KnownType.cs [hunk 1]
```diff
@@ -401,6 +401,7 @@ internal sealed partial class KnownType
         internal static readonly KnownType System_Security_Cryptography_DSACryptoServiceProvider = new("System.Security.Cryptography.DSACryptoServiceProvider");
         internal static readonly KnownType System_Security_Cryptography_ECDiffieHellman = new("System.Security.Cryptography.ECDiffieHellman");
         internal static readonly KnownType System_Security_Cryptography_ECDsa = new("System.Security.Cryptography.ECDsa");
+        internal static readonly KnownType System_Security_Cryptography_ECAlgorythm = new("System.Security.Cryptography.ECAlgorithm");
```
Comment (cristian-ambrosini-sonarsource) [inline analyzers/src/SonarAnalyzer.Common/Helpers/KnownType.cs:404]: ECAlgorithm is a common base type for ECDsa and ECDiffieHellman introduced in .Net 7 (link to the [proposal](https://github.com/dotnet/runtime/issues/61516) and the related [PR](https://github.com/dotnet/runtime/pull/61753))

File: analyzers/tests/SonarAnalyzer.UnitTest/TestCases/CryptographicKeyShouldNotBeTooShort.CSharp9.cs [hunk 2]
```diff
@@ -72,6 +72,6 @@ public void GenerateKey()
         ec2.GenerateKey(ECCurve.NamedCurves.brainpoolP160t1); // Noncompliant {{Use a key length of at least 224 bits for EC cipher algorithm.}}
 
         ECDsaOpenSsl ec3 = new();
-        ec3.GenerateKey(ECCurve.NamedCurves.brainpoolP192t1); // FN, See: https://github.com/SonarSource/sonar-dotnet/issues/6164
+        ec3.GenerateKey(ECCurve.NamedCurves.brainpoolP192t1); // Noncompliant {{Use a key length of at least 224 bits for EC cipher algorithm.}}
```
Comment (cristian-ambrosini-sonarsource) [inline analyzers/tests/SonarAnalyzer.UnitTest/TestCases/CryptographicKeyShouldNotBeTooShort.CSharp9.cs:75]: ECDsaOpenSsl.BaseType.ConstructedFrom 
_Before .Net 7_
Crypthography.ECDsaOpenSsl (which baseType is **Crypthography.ECDsa**)
_From .Net 7_
**Crypthography.ECAlgorythm**

File: analyzers/src/SonarAnalyzer.CSharp/Rules/CryptographicKeyShouldNotBeTooShort.cs [hunk 3]
```diff
@@ -220,7 +221,7 @@ private static void CheckBouncyCastleParametersGenerators(ITypeSymbol containing
```
Comment (martin-strecker-sonarsource) [inline analyzers/src/SonarAnalyzer.CSharp/Rules/CryptographicKeyShouldNotBeTooShort.cs:221]: Line 104 and 105 can be joined https://github.com/SonarSource/sonar-dotnet/blob/e84d7a22fc362fa4a582c9869fb81cb81c34ab2c/analyzers/src/SonarAnalyzer.CSharp/Rules/CryptographicKeyShouldNotBeTooShort.cs#L103-L106

```cs
 if (GetPropertyName(assignment.Left) == "KeySize" 
     && assignment.Left is MemberAccessExpressionSyntax { Expression: { }  expression }
     && c.SemanticModel.GetTypeInfo(expression).Type is ITypeSymbol containingType) 
```

Comment (martin-strecker-sonarsource) [reply analyzers/src/SonarAnalyzer.CSharp/Rules/CryptographicKeyShouldNotBeTooShort.cs:221]: There are some property names hardcoded in the file like `"KeySize"`. Please check if `nameof` can be used instead.

**Review by martin-strecker-sonarsource — CHANGES_REQUESTED** [review]
Looks good to me. I think you can do some polishing in the file while you are at it.

**Review by martin-strecker-sonarsource — APPROVED** [review]
LGTM.

There might be another issue with these lines:

https://github.com/SonarSource/sonar-dotnet/blob/b7d436ca7b28ae1b02392af9b146121fb9f0df17/analyzers/src/SonarAnalyzer.CSharp/Rules/CryptographicKeyShouldNotBeTooShort.cs#L235-L239

It is probably working correctly, but we don't have it under test:
```cs
 ECDsaOpenSsl ec3 = new();
  ec3?.GenerateKey();
  ec3!.GenerateKey();
```
Same for the KeySize property.

Can you add the four test cases, please? You can do this in this PR or in a follow-up.

**Pull request merged**